src/Common/tests/Tests/System/StringTests.Span.cs

```diff
@@ -0,0 +1,1317 @@
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdefghij_lmn", StringComparison.Ordinal, false)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "a", StringComparison.Ordinal, true)]
+        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyza", StringComparison.Ordinal, false)]
+        // OrdinalIgnoreCase
+        [InlineData("Hello", "Hel", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "Hello", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "Hello Larger", StringComparison.OrdinalIgnoreCase, false)]
+        [InlineData("Hello", "", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "HEL", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("Hello", "Abc", StringComparison.OrdinalIgnoreCase, false)]
+        [InlineData("Hello", SoftHyphen + "Hel", StringComparison.OrdinalIgnoreCase, false)]
+        [InlineData("", "", StringComparison.OrdinalIgnoreCase, true)]
+        [InlineData("", "hello", StringComparison.OrdinalIgnoreCase, false)]
+        public static void StartsWith(string s, string value, StringComparison comparisonType, bool expected)
+        {
+            if (comparisonType == StringComparison.CurrentCulture)
+            {
+                Assert.Equal(expected, s.StartsWith(value));
+            }
+            Assert.Equal(expected, s.AsSpan().StartsWith(value.AsSpan(), comparisonType));
+        }
+
+        [Theory]
+        [ActiveIssue("https://github.com/dotnet/coreclr/issues/2051", TestPlatforms.AnyUnix)]
+        [InlineData(StringComparison.CurrentCulture)]
+        [InlineData(StringComparison.CurrentCultureIgnoreCase)]
+        [InlineData(StringComparison.Ordinal)]
+        [InlineData(StringComparison.OrdinalIgnoreCase)]
+        public static void StartsWith_NullInStrings(StringComparison comparison)
+        {
+            Assert.False("\0test".AsSpan().StartsWith("test".AsSpan(), comparison));
+            Assert.False("te\0st".AsSpan().StartsWith("test".AsSpan(), comparison));
+            Assert.True("te\0st".AsSpan().StartsWith("te\0s".AsSpan(), comparison));
+            Assert.True("test\0".AsSpan().StartsWith("test".AsSpan(), comparison));
+            Assert.False("test".AsSpan().StartsWith("te\0".AsSpan(), comparison));
+        }
+
+        [Theory]
+        [InlineData("Hello", 0, 5, "Hello")]
+        [InlineData("Hello", 0, 3, "Hel")]
+        [InlineData("Hello", 2, 3, "llo")]
+        [InlineData("Hello", 5, 0, "")]
+        [InlineData("", 0, 0, "")]
+        public static void Substring(string s, int startIndex, int length, string expected)
+        {
+            if (startIndex + length == s.Length)
+            {
+                Assert.Equal(expected, s.Substring(startIndex));
+                Assert.Equal(expected, s.AsSpan(startIndex).ToString());
+            }
+            Assert.Equal(expecte
```

Comment:
@ahsonkhan should i gruop inputs on bottom?

Reply:
I don't understand the question here.

Reply:
Sure. I would be fine with organizing the MemberData args this way.